common/worker_versioning/worker_versioning.go

```diff
@@ -74,3 +81,19 @@ func FindBuildId(versionSets []*taskqueuepb.CompatibleVersionSet, buildId string
 	}
 	return setIndex, indexInSet
 }
+
+func WorkflowsExistForBuildId(ctx context.Context, visibilityManager manager.VisibilityManager, ns *namespace.Namespace, taskQueue, buildId string) (bool, error) {
+	escapedTaskQueue := sqlparser.String(sqlparser.NewStrVal([]byte(taskQueue)))
+	escapedBuildId := sqlparser.String(sqlparser.NewStrVal([]byte(VersionedBuildIdSearchAttribute(buildId))))
+	query := fmt.Sprintf("%s = %s AND %s = %s", searchattribute.TaskQueue, escapedTaskQueue, searchattribute.BuildIds, escapedBuildId)
```

Comment:
I'm debating if we want to filter by task queue or not.
We might not want to delete build ids if there are any associated open workflows on any task queues.
The reason is we might have cross task queue compatible workloads and we'll need the build ids to prevent tasks from getting stuck.

Looks like cross task queue workloads are going to need some more thought.